scripts/MCNPMesh.gd

```diff
@@ -0,0 +1,178 @@
+extends Spatial
+
+# arrays that contain the boundaries of voxels 
+var bounds_x = []
+var bounds_y = []
+var bounds_z = []
+
+# array that contains voxel centroids and radiation values
+# index 0 is a Vector3 of the centroid position
+# index 1 is radiation value
+# index 2 is relative error
+var radiation_map = []
+
+# maximum indices for the 3 axes and for radiation_map
+var max_index_x = 0
+var max_index_y = 0
+var max_index_z = 0
+var max_index = 0
+
+# index of the last voxel queried from getRadiation()
+var prev_index = -1
+
+# bool for whether the y or z axis is considered the vertical axis in the MCNP file
+var y_is_up = true
+
+# Used to get the x,y,z boundaries of the voxels in the radiation map
+# units is the units of the boundaries. (Ex. if boundaries are in cm, units 
+# 	would be 1/100)
+func get_bounds(csv_line, units):
+	var bounds = []
+	for i in range(0, csv_line.size()):
+		bounds.append(float(csv_line[i]) * units); 
+	return bounds
+
+# Initializes the MCNP mesh. 
+# This should be called ONCE in the child class's _ready() function.
+# y_is_up_IN should be true if the MCNP file designates y as the vertical axis
+#		and false if it designates z as the vertical axis.
+func initialize(csv_file_name, y_is_up_IN, units_IN):	
+	var csv_file = File.new()
+	csv_file.open(csv_file_name, File.READ)
+	
+	# set internal variables
+	y_is_up = y_is_up_IN
+	var units = units_IN
+	
+	# get x,y,z boundaries
+	bounds_x = get_bounds(csv_file.get_csv_line(), units)
+	if y_is_up:
+		bounds_y = get_bounds(csv_file.get_csv_line(), units)
+		bounds_z = get_bounds(csv_file.get_csv_line(), units)
+	else:
+		bounds_z = get_bounds(csv_file.get_csv_line(), units)
+		bounds_y = get_bounds(csv_file.get_csv_line(), units)
+		
+	# calculate max x,y,z indices 
+	# (-1) there's one extra bound compared to number of spaces
+	max_index_x = bounds_x.size() - 1
+	max_index_y = bounds_y.size() - 1
+	max_index_z = bounds_z.size() - 1
+	
+	# get voxel data
+	if y_is_up:
```

Comment:
What about...
```
if y_is_up:
   up_index = 1
   side_index = 2
else:
   up_index = 2
   side_index = 1
```

And then you only need one version of the below with:

```
        voxel_centroid.y = float(csv_line[up_index]) * units
        voxel_centroid.z = float(csv_line[side_index]) * units
```

Reply:
Good idea, I will add it!